LibraBFT/Concrete/Properties/VotesOnce.agda

```diff
@@ -36,35 +35,59 @@ open import LibraBFT.Yasm.Yasm ℓ-RoundManager ℓ-VSFP ConcSysParms PeerCanSig
 -- sent by honest peers in the implementation, then the implemenation
 -- satisfies the LibraBFT.Abstract.Properties.VotesOnce invariant.
 
-module LibraBFT.Concrete.Properties.VotesOnce where
+module LibraBFT.Concrete.Properties.VotesOnce (𝓔 : EpochConfig) where
  -- TODO-3: This may not be the best way to state the implementation obligation.  Why not reduce
  -- this as much as possible before giving the obligation to the implementation?  For example, this
  -- will still require the implementation to deal with hash collisons (v and v' could be different,
  -- but yield the same bytestring and therefore same signature).  Also, avoid the need for the
  -- implementation to reason about messages sent by step-cheat, or give it something to make this
  -- case easy to eliminate.
 
- ImplObligation₁ : Set (ℓ+1 ℓ-RoundManager)
- ImplObligation₁ =
+ record VoteForRound∈ (pk : PK)(round : ℕ)(epoch : ℕ)(bId : HashValue)(pool : SentMessages) : Set where
+   constructor mkVoteForRound∈
+   field
+     msgWhole  : NetworkMsg
+     msgVote   : Vote
+     msg⊆      : msgVote ⊂Msg msgWhole
+     msgSender : ℕ
+     msg∈pool  : (msgSender , msgWhole) ∈ pool
+     msgSigned : WithVerSig pk msgVote
+     msgEpoch≡ : msgVote ^∙ vEpoch ≡ epoch
+     msgRound≡ : msgVote ^∙ vRound ≡ round
+     msgBId≡   : msgVote ^∙ vProposedId ≡ bId
+ open VoteForRound∈ public
+
+
+ NewVoteSignedAndRound>0 : Set (ℓ+1 ℓ-RoundManager)
```

Comment:
This is not used.  Instead we have an equivalent `postulate` called `¬genVotesRound≢0` in `LibraBFT.Impl.Properties.Handle`, which /is/ used in the concrete proofs (I think this is something I did at some point, so I take responsibility :smile:.)  The concrete proofs should not be reaching into a particular implementation for properties.  So I suggest removing the postulate mentioned above, and promoting `NewVoteSignedAndRound>0` to an implementation obligation.

Reply:
@lisandrasilva are you going to address this one?

Reply:
@lisandrasilva are you going to address this one?